File: android/src/main/java/com/reactnativechimesdk/itf/SimpleVideoTileObserver.java
package com.reactnativechimesdk.itf;

import android.util.Log;

import androidx.annotation.NonNull;

import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileObserver;
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileState;

public interface SimpleVideoTileObserver extends VideoTileObserver {

  String TAG = "SimpleVideoTileObserver";

  @Override
  default void onVideoTileAdded(@NonNull VideoTileState videoTileState) {
    Log.d(TAG, "onVideoTileAdded: " + videoTileState.getTileId());
  }

  @Override
  default void onVideoTilePaused(@NonNull VideoTileState videoTileState) {

  }

  @Override
  default void onVideoTileRemoved(@NonNull VideoTileState videoTileState) {
    Log.d(TAG, "onVideoTileRemoved: " + videoTileState.getTileId());
  }

  @Override
  default void onVideoTileResumed(@NonNull VideoTileState videoTileState) {

  }

  @Override
  default void onVideoTileSizeChanged(@NonNull VideoTileState videoTileState) {

  }
}

File: android/src/main/java/com/reactnativechimesdk/itf/SimpleVideoTileObserver.kt
package com.reactnativechimesdk.itf

import android.util.Log
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileObserver
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileState

interface SimpleVideoTileObserver : VideoTileObserver {

  companion object {
    const val TAG = "SimpleVideoTileObserver"
  }

  override fun onVideoTileAdded(tileState: VideoTileState) {
    Log.d(TAG, "onVideoTileAdded: " + tileState.attendeeId)
  }

  override fun onVideoTilePaused(tileState: VideoTileState) {
  }

  override fun onVideoTileRemoved(tileState: VideoTileState) {
    Log.d(TAG, "onVideoTileRemoved: ")
  }

  override fun onVideoTileResumed(tileState: VideoTileState) {
  }

  override fun onVideoTileSizeChanged(tileState: VideoTileState) {
  }
}
